extensions/amp-sharethrough/0.1/amp-sharethrough.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {AmpMustache} from '../../../build/all/v0/amp-mustache-0.1.max';
+import {xhrFor} from '../../../src/xhr';
+
+/** @private @const */
+/**
+ * @type {string} The address where ads are fetched from
+ */
+const serverUrl = 'https://btlr.sharethrough.com/';
+/**
+ * @type {string} The address where analytics are posted
+ */
+const analyticsUrl = 'https://b.sharethrough.com/';
+
+/**
+ * @param {!Window} win Destination window for the new element.
+ * @this {undefined}  // Make linter happy
+ * @return {undefined}
+ */
+export class AmpSharethrough extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * Prefetches and preconnects URLs related to the ad.
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(serverUrl, onLayout);
+    this.preconnect.url(analyticsUrl, onLayout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const pkey = AMP.assert(this.element.getAttribute('data-native-key'),
+      'The data-native-key attribute is required for <amp-sharethrough> %s',
+      this.element);
+
+    const url = `${serverUrl}v4?placement_key=${pkey}`;
+
+    return xhrFor(this.getWin()).fetchJson(url)
+      .then(data => {
+        if (data.creatives.length > 0) {
+
+          const creative = data.creatives[0].creative;
+          const rawTemplate = data.placement.placementAttributes.template;
+
+          const templateElement = document.createElement('div');
+          templateElement./*REVIEW*/innerHTML = this.unencodeHTML(rawTemplate);
```

Comment:
@dvoytenko Thanks for the feedback. Here's an example of the `rawTemplate` and in our system it's always a mustache template. 

```
<div class="{{action}} str-collapsed" style="background-color: #f6f6f6 ; overflow:hidden; height:auto; clear:both;cursor:pointer;">
  <a href="{{media_url}}">
  <div class="str-thumbnail" style="width:90px; height:90px; float:left;margin-right:8px;">
    <amp-img src="{{{thumbnail_url}}}" layout="fixed" height="80" width="80" class="str-thumbnail"></amp-img>
  </div>
  <h3 class="str-title headline" style="margin: 0;padding: 2px 3px 0 0;font-weight:bold;line-height:normal;">{{title}}</h3>
  <div class="str-description" style="margin: 0;padding: 2px 3px 0 0;line-height:normal;">{{description}}</div>
  <div class="str-advertiser" style="margin:0; padding 4px 3px 4px 0;font-size: .8em;padding: 4px 4px 4px 0;">Ad by {{advertiser}}
    <div class="str-brand-logo" style="display: inline-block;width:16px; height:16px; max-width: 16px; max-height: 16px;margin-left: 6px;">
        <amp-img src="{{{brand_logo_url}}}" layout="fixed" height="16" width="16"></amp-img>
    </div>
  </div>
  </a>
</div>
```


I'm not exactly sure what you mean by compiled into the binary? In the XHR response we get is json that contains the mustache template and other values that we then insert into the template to create a rendered template which we then insert into the DOM.